Add user system001

MoltndarKen opened the pull request:
# WHAT

Implement user systems
- login
- logout
- authenticate_user
- haml viewfiles for users
- show routing for user(associeting with prototypes)
- add Profile column
  # WHY
  to Implement user systems

Commit:
# WHAT
run 'rails g devise:install'
#WHY
to generate user model

Commit:
# WHAT
generate model, and users tables by using devise

Commit:
# WHAT
add column name,work,member,github to users table

Commit:
#WHAT
describe association between user and prototypes
#WHY
to display index,show screen

Commit:
# WHAT
generate user view files by running 'rails g devise:views users'

Commit:
# WHAT
add gem file 'gem 'bootstrap-sass' and add application.css.scss '@import "bootstrap-sprockets";'
'
# WHY
improve  SASS::SyntaxError

Commit:
# WHAT
change the view files
# WHY
to describe user routing as RESTful routing

Commit:
#WHAT
generate users controller
# WHY
to designate actions

Commit:
#WHAT
routing user and define actions in users_controller
#WHY
to  make MVC about user
#problems
If I access 'users/sign_in', chrome display users#show!
I define users/sign_in in route.rb, then,problem was improved

Commit:
# WHAT
add column 'Profile' to users table

Commit:
# WHAT
routing users by using 'devise_for' method
# WHY
couldn't use devise helper method(user_signed_in?) because of not using 'devise_for' method

Commit:
# WHAT
 add authenticate_user!  in application_controller and check  whether the  it is reflected  then, routing prototype#new
# WHY
 to limit the function when you are not logged in

Commit:
# WHAT
 run 'rails g devise:views'

Commit:
# WHAT
this is mistaked change, then , I' reset this commit

Commit:
Revert "# WHAT"

This reverts commit 0c17d3a46de1a0bcd6a2353dd45d20ee88a97ee0.

Commit:
Merge branch 'add_user_system001' of https://github.com/MoltndarKen/protospace into add_user_system001

Commit:
# WHAT
Cancel the commit that was pushed to the remote
# WHY
mistaken commit  which name is '86bb93216208dbcf96255c46189886563dff3c8d'
'# WHAT
this is mistaked change, then , I' reset this commit'

Commit:
# WHAT
replace devise html.erb to html.haml by using ‘html2haml’ gem

Commit:
# WHAT
Replace the view files in the marked-up file

Commit:
# WHAT
edit view file by using form_for method

Commit:
# WHAT
add name attribute to sign_in view file
# WHY
to get parameter

Commit:
# WHAT
edit login view file by using form_for method
# why
to Implement the login function

Commit:
# WHAT
implement login,sign_out button by using user_signed_in? method

Commit:
# WHAT
delete unnecessary files

Commit:
# WHAT
Put a line break

Commit:
# WHAT
check dropdown menu can move but it can't be worked because of not adopt bootstrap files

Commit:
# WHAT
edit show pages (github,and so on)

Commit:
#  WHAT
 fix application controller about 'after_sign_out_path_for' ,'after_sign_in_path_for'
# WHY
don't run above those methods . when I write those method in ApplicationController Class , this method ran.

Commit:
# WHAT
fix design

Commit:
# WHAT
Put a blank line

Commit:
Put a blank line

ogqwa commented on Gemfile:
```diff
@@ -44,8 +44,10 @@ gem 'devise'
-gem 'carrierwave' 
+gem 'carrierwave'
+gem 'html2haml'
+gem 'rmagick'
```
空行を入れましょう

ogqwa commented on app/assets/javascripts/users.js.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
使わないファイルは消してしまいましょう！

ogqwa commented on app/views/devise/registrations/edit.html.haml:
```diff
@@ -0,0 +1,37 @@
+        .user-name_image.row
+          .user-name.col-md-10
+            %input{:type => "text", :placeholder => "Username", :value => "TECH::MASTER"}/
+          .user-image.col-md-2{:style => 'background-image: url(http://泣ける動画.com/wp-content/uploads/2015/01/t0205-01.jpg);'}
```
新しい記法で書きましょう１

ogqwa commented on app/views/layouts/_header.html.haml:
```diff
@@ -10,20 +10,31 @@
+            .btn.btn-primary.navbar-btn
+              = link_to "Logout", destroy_user_session_path, method: :delete
+            .btn.btn-primary.navbar-btn
+              %a{:href => "/users/#{current_user.id}"}
```
新しい記法で書きましょう

ogqwa commented on app/views/layouts/application.html.haml:
```diff
@@ -8,5 +8,6 @@
   %body
     = render "layouts/header"
-    = yield
+    .dower{style: "margin-top:30px;"}
```
cssは外部ファイルに記述するようにしましょう

ogqwa commented on app/views/prototypes/latest/_latest.html.haml:
```diff
@@ -1,5 +1,19 @@
-%br
+.col-sm-4.col-md-3.proto-content
+  .thumbnail
+    %a{:href => ""}
```
新しい記法で書きましょう

ogqwa commented on app/views/prototypes/latest/index.html.haml:
```diff
@@ -1,3 +1,37 @@
+.text-center
+  %ul.pagination
+    %li.disabled
+      %a{"aria-label" => "Previous", :href => "#"}
```
新しい記法で書きましょう

ogqwa commented on app/views/prototypes/ranking/_ranking.html.haml:
```diff
@@ -1,5 +1,19 @@
-%br
+.col-sm-4.col-md-3.proto-content
+  .thumbnail
+    %a{:href => ""}
```
新しい記法で書きましょう

ogqwa commented on Gemfile:
```diff
@@ -44,8 +44,10 @@ gem 'devise'
-gem 'twitter-bootstrap-rails'
+gem 'bootstrap-sass'
 gem 'bootstrap-material-design', '0.1.4'
 gem 'slim-rails'
```
今回はhamlを使用するので消してしまいましょう

Commit:
# what
adapt new notation of hash and delete unnecessary files and apply the external css file (assets/stylesheets/custom/main.css)

ogqwa commented on app/assets/stylesheets/custom/main.css:
```diff
@@ -0,0 +1,3 @@
+.dower{
+  margin-top:30px;
+}
```
空行を入れましょう

ogqwa commented on test/fixtures/users.yml:
```diff
@@ -0,0 +1,11 @@
+# column: value
+#
+two: {}
+#  column: value
```
不要なので消してしまいましょう

ogqwa commented on practice.html.haml:
```diff
@@ -0,0 +1,43 @@
+-#   .text-center
+    %p
+      Not a member?
+      %a{:href => "/users/sign_up"} Sign up now
```
新しい記法に書き換えましょう

ogqwa commented on app/views/users/show.html.haml:
```diff
@@ -0,0 +1,64 @@
+                  =prototype.created_at.strftime("%Y/%m/%d(#{%w(日 月 火 水 木 金 土)[prototype.created_at.wday]})") 
+              %ul.proto-tag-list.list-inline
+                %li
+                  %a{:href => "#", :class => "btn btn-default"} iPad
```
新しい記法に書き換えましょう

ogqwa commented on app/views/devise/sessions/new.html.haml:
```diff
@@ -0,0 +1,23 @@
+        = f.label :email, 'Email Address'
+      #inputPassword.form-control
+        =f.email_field :email, autofocus: true, :placeholder =>'Email Address'
+      .sr-only{:for => "inputEmail"}
```
新しい記法に書き換えましょう

ogqwa commented on app/views/devise/registrations/new.html.haml:
```diff
@@ -0,0 +1,34 @@
+          %h4 Works
+          .user-name_image.row
+            .user-name.col-md-12
+              %textarea{:name => "work", :cols => "30", :rows => "4", :placeholder => "Works", style: "background-color:white;"}
```
新しい記法に書き換えましょう

ogqwa commented on app/views/devise/registrations/edit.html.haml:
```diff
@@ -0,0 +1,37 @@
+        .user-name_image.row
+          .user-name.col-md-10
+            %input{type: "text", placeholder: "Username", value: "TECH::MASTER"}/
+          .user-image.col-md-2{:style => 'background-image: url(http://泣ける動画.com/wp-content/uploads/2015/01/t0205-01.jpg);'}
```
新しい記法に書き換えましょう

Commit:
# what&why
replace new notificasion about hash and delete unnecesarry files

ogqwa commented on app/views/devise/sessions/new.html.haml:
```diff
@@ -0,0 +1,22 @@
+  .text-center
+    %p
+      Not a member?
+      %a{href: "/users/sign_up"} Sign up now
```
空行入れましょう

ogqwa commented:
空行を入れるべきところに入れたらLGTMで！

Commit:
# WHAT&WHY
place a blank line

MoltndarKen commented:
はい！ありがとうございます^^
